### PR TITLE
✨ Document new gated repository features

### PR DESCRIPTION
**Pull request by SBrandeis**
In preparation for https://github.com/huggingface/moon-landing/pull/5179 (internal link)

### CODE REVIEW

File: docs/hub/datasets-gated.md [hunk 1]
```diff
@@ -21,6 +25,19 @@ extra_gated_fields:
 
 The `README.md` file for a dataset is called a [Dataset Card](./datasets-cards). Visit the documentation to learn more about how to use it and to see the properties that you can configure.
 
+## Manual approval
+
+By default, users request to accept the dataset are accepted automatically.
```
Comment (julien-c) [inline docs/hub/datasets-gated.md:30]:
```suggestion
By default, users requests to access the dataset are accepted automatically.
```

Comment (SBrandeis) [reply docs/hub/datasets-gated.md:30]: thx!

File: docs/hub/datasets-gated.md [hunk 2]
```diff
@@ -27,10 +27,10 @@ The `README.md` file for a dataset is called a [Dataset Card](./datasets-cards).
 
 ## Manual approval
 
-By default, users request to accept the dataset are accepted automatically.
+By default, users requests to access the dataset are accepted automatically.
```
Comment (Michellehbn) [inline docs/hub/datasets-gated.md:30]:
```suggestion
By default, requests to access the dataset are automatically accepted.
```

File: docs/hub/datasets-gated.md [hunk 3]
```diff
@@ -27,10 +27,10 @@ The `README.md` file for a dataset is called a [Dataset Card](./datasets-cards).
...
 Dataset authors can set the approval mode to "Manual reviews" from the dataset's **Settings** tab.
 Doing so enforces that each access request is manually reviewed and approved by the dataset authors.
```
Comment (Michellehbn) [inline docs/hub/datasets-gated.md:32]:
```suggestion
Doing so enforces that each access request will be manually reviewed and approved by the dataset authors.
```

File: docs/hub/models-gated.md [hunk 4]
```diff
@@ -44,10 +44,10 @@ The `README.md` file for a model is called a [model Card](./model-cards). Visit
 
 ## Manual approval
 
-By default, users request to accept the model are accepted automatically.
+By default, users requests to access the model are accepted automatically.
```
Comment (Michellehbn) [inline docs/hub/models-gated.md:47]:
```suggestion
By default, requests to access the model are automatically accepted.
```

File: docs/hub/models-gated.md [hunk 5]
```diff
@@ -44,10 +44,10 @@ The `README.md` file for a model is called a [model Card](./model-cards). Visit
...
 Model authors can set the approval mode to "Manual reviews" from the model's **Settings** tab.
 Doing so enforces that each access request is manually reviewed and approved by the model authors.
```
Comment (Michellehbn) [inline docs/hub/models-gated.md:49]:
```suggestion
Doing so enforces that each access request will be manually reviewed and approved by the model authors.
```

**Review by Michellehbn — APPROVED** [review]
made minor grammar recommendations! looks great!

**Pull request merged**